scripts/unit-tests/windows/ovs-build.ps1

```diff
@@ -0,0 +1,408 @@
+}
+
+function Start-SSHCMD ($server, $user, $key, $cmd) {
+    write-host "Running ssh command $cmd on remote server $server"
+#        &plink -batch $server -l $user -i $key $cmd
+    echo Y | plink.exe $server -l $user -i $key $cmd
+}
+
+function Start-SCP ($server, $user, $key, $localPath, $remotePath) {
+    write-host "Starting copying $localPath to remote location ${server}:${remotePath}"
+#        &pscp -batch -scp -r -i $key $localPath $user@${server}:${remotePath}
+    echo Y | pscp.exe -scp -r -i $key $localPath $user@${server}:${remotePath}
+}
+
+function Start-CreateRemoteDir {
+    Set-RemoteVars
+    $logCMD = "mkdir -p $remotelogDir/$commitID"
+    $msiCMD = "mkdir -p $remotemsiDir/$commitID"
+    Start-SSHCMD $server $user $key $logCMD
+    Start-SSHCMD $server $user $key $msiCMD
+}
+
+function Start-RemoteLogCopy ($locallogPath) {
+    Set-RemoteVars
+    $remotelogPath = "$remotelogDir/$commitID"
+    write-host "Started copying logs for commit ID $commitID unit tests to remote location ${server}:${remotelogPath}"
+    Start-SCP $server $user $key $locallogPath $remotelogPath
+}
+
+function Start-RemoteMsiCopy ($localmsiPath) {
+    Set-RemoteVars
+    $remotemsiPath = "$remotemsiDir/$commitID"
+    write-host "Started copying generated msi for commit ID $commitID to remote location ${server}:${remotemsiPath}"
+    Start-SCP $server $user $key $localmsiPath $remotemsiPath
+}
+
+function Start-Cleanup {
+    $ovsprocess = Get-Process ovs* -ErrorAction SilentlyContinue
+    $ovnprocess = Get-Process ovn* -ErrorAction SilentlyContinue
+    if ($ovsprocess) {
+        Stop-Process -name ovs*
+    }
+    if ($ovnprocess) {
+        Stop-Process -name ovn*
+    }
+    cd $buildDir
+    CheckRemoveDir $commitDir
+}
+
+function Start-MailJob ($user, $pass, $status) {
+    $uri = "http://10.20.1.3:8080/job/ovs-email-job/buildWithParameters?token=b204eee759ab38ebb986d223f6c5b4ce&commitid=$commitID&status=$status"
+    $pair = "${user}:${pass}"
+    $bytes = [System.Text.Encoding]::ASCII.GetBytes($pair)
+    $base64 = [System.Convert]::ToBase64String($bytes)
+    $basicAuthValue = "Basic $base64"
+    $headers = @{ Authorization = $basicAuthValue }
+    write-host "Starting the Mail Job for status $status"
+    Invoke-WebRequest -uri $uri -Headers $headers -UseBasicParsing
+}
+
+function Set-MakeSh {
+$makeScript = @"
+#!/bin/bash
+set -e
+cd $msysCwd
+echo `$INCLUDE
+./boot.sh
+./configure CC=./build-aux/cccl LD="C:/Program Files (x86)/Microsoft Visual Studio 12.0/VC/BIN/x86_amd64/link.exe" LIBS="-lws2_32 -liphlpapi -lwbemuuid -lole32 -loleaut32" --prefix="C:/ProgramData/openvswitch" \
+--localstatedir="C:/ProgramData/openvswitch" --sysconfdir="C:/ProgramData/openvswitch" \
+--with-pthread="$pthreadDir" --with-vstudiotarget="Debug"
+make clean && make -j4
+exit `$?
+"@
+
+$makeScriptPath = Join-Path $pwd "make.sh"
+$makeScript.Replace("`r`n","`n") | Set-Content $makeScriptPath -Force
+return $makeScriptPath
+}
+
+function Set-UnitSh {
```

Comment:
I left them here instead of a file for the simplicity. There would be 3 different file that would have to be copied into the build folder prior to running them. I thought this way was cleaner and simpler